src/connect.rs

```diff
@@ -826,13 +841,17 @@ mod socks {
                 &password,
             )
             .await
+            .map_err(|_| crate::error::builder("failure connecting socks"))?
```

Comment:
huh, I'll double check this, you might be right.

Reply:
Ok, so, if i remove the `map_err` in this line, I get

```
error[E0277]: the trait bound `tokio_socks::error::Error: std::error::Error` is not satisfied
   --> src/connect.rs:843:19
    |
843 |             .await?
    |                   ^ the trait `std::error::Error` is not implemented for `tokio_socks::error::Error`
    |
    = note: required because of the requirements on the impl of `std::convert::From<tokio_socks::error::Error>` for `std::boxed::Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>`
    = note: required by `std::convert::From::from`
```

Admittedly, `tokio-socks` should probably `impl std::error::Error` on their Error type. But as long as that is not the case, I guess we need the `map_err` here, right?

Reply:
I confess, I had just used `crate::error::builder` because 
1. I saw it somewhere else in the codebase and 
2. it worked

thinking that it was some kind of builder for stringly-typed errors. What Error kind would be correct here, I'm a bit lost...

Reply:
Oh bummer, tokio-socks uses `failure` (sigh, I really wish people would stop using it in libraries). It's not the best, but the error case shouldn't be frequent, so this is fine: just convert it to a string.

```rust
.map_err(|e| format!("socks connect error: {}", e))?
```

Reply:
changed